core/src/main/java/com/alibaba/nacos/core/utils/WebUtils.java

```diff
@@ -71,4 +72,20 @@ public static String getAcceptEncoding(HttpServletRequest req) {
         encode = encode.contains(",") ? encode.substring(0, encode.indexOf(",")) : encode;
         return encode.contains(";") ? encode.substring(0, encode.indexOf(";")) : encode;
     }
+
+    /**
+     * Returns the value of the request header "user-agent" as a <code>String</code>.
+     *
+     * @param request HttpServletRequest
+     * @return the value of the request header "user-agent", or the value of the
+     *         request header "client-version" if the request does not have a
+     *         header of "user-agent"
+     */
+    public static String getUserAgent(HttpServletRequest request) {
```

Comment:
I think I should is to check the Client - Version first, then the user-agent

Reply:
> I think I should is to check the Client - Version first, then the user-agent

If "Client-Version" is checked first, "Client-Version" is still used as "User-Agent". In the HttpSimpleClient#setHeaders method, "Client-Version" is assigned ${project.version} and is not done. To the "Client-Version" unified, because "User-Agent" = NacosClientName + ":v" + ${project.version};, the two request headers are not all requests are the same

Check "User-Agent" first, can get the correct "User-Agent", but also compatible with the old code